src/installer/pkg/sfx/installers/dotnet-host.proj

```diff
@@ -78,4 +78,36 @@
       <RpmJsonProperty Include="@(PackageConflictsProperty)" />
     </ItemGroup>
   </Target>
+
+  <PropertyGroup>
+    <_MacOSIntermediatesPath>$(IntermediateOutputPath)macos/</_MacOSIntermediatesPath>
+    <!-- CreatePkg depends on this property being set.-->
+    <MacOSScriptsDirectory>$(_MacOSIntermediatesPath)scripts</MacOSScriptsDirectory>
+  </PropertyGroup>
+  <ItemGroup>
+    <_MacOSScript Include="$(MacOSScriptsTemplateDirectory)/*" Destination="$(MacOSScriptsDirectory)/%(FileName)%(Extension)"/>
+  </ItemGroup>
+  <Target Name="ReplaceTemplateParametersInMacOsScripts" 
+          Inputs="@(_MacOSScript)"
+          Outputs="@(_MacOSScript->'%(Destination)')"
+          DependsOnTargets="_GetInstallerProperties"
+          BeforeTargets="CreatePkg">
+
+    <PropertyGroup>
+      <_UnameMachineRegex>$(InstallerTargetArchitecture)</_UnameMachineRegex>
+      <_UnameMachineRegex Condition="'$(InstallerTargetArchitecture)' == 'arm64'">arm64|aarch64</_UnameMachineRegex>
+      <_UnameMachineRegex Condition="'$(InstallerTargetArchitecture)' == 'x64'">amd64|x86_64</_UnameMachineRegex>
+      <_MacOSScriptsTemplateProperties>InstallerTargetArchitecture=$(InstallerTargetArchitecture);UnameMachineRegex=$(_UnameMachineRegex)</_MacOSScriptsTemplateProperties>
+    </PropertyGroup>
+    
+    <GenerateFileFromTemplate TemplateFile="@(_MacOSScript)" 
```

Comment:
This is using the copy of this target that's on the referenced package right? Most of the repo doesn't do that https://github.com/dotnet/runtime/blob/360df71eadde0d1394ee7b89693f83913f75575d/src/installer/pkg/sfx/Microsoft.NETCore.App/Directory.Build.targets#L16

vs I think this one comes from https://github.com/dotnet/arcade/blob/139794062d986f8984bf06f0cd564ac4dbbebada/src/Microsoft.DotNet.Build.Tasks.Templating/src/build/Microsoft.DotNet.Build.Tasks.Templating.props#L8. Can you add a comment to make it obvious please?

Reply:
I believe that should be clear due to the `PackageReference` in the same project.  Long term we shouldn't be building custom infra in dotnet/runtime, so those other places should move to use this.  This change is just a cherry-pick of the commit which was already made in dotnet/runtime.  We shouldn't be churning it in release.

Reply:
I opened https://github.com/dotnet/runtime/issues/59355 to track that.